### PR TITLE
feat: mark building as merged

### PR DESCRIPTION
**Pull request by emalfroy**

### CODE REVIEW

File: src/BuildingRegistry/Building/Building.cs [hunk 1]
```diff
@@ -236,6 +236,26 @@ public void ChangeOutliningConstruction(ExtendedWkbGeometry extendedWkbGeometry)
...
+                         .Concat(_buildingUnits.RealizedBuildingUnits())
+                         .Where(x => x.Function != BuildingUnitFunction.Common)
+                         .ToList())
+            {
+                ApplyChange(new BuildingUnitWasMoved(
+                    BuildingPersistentLocalId,
+                    buildingUnit.BuildingUnitPersistentLocalId,
+                    destinationBuildingPersistentLocalId));
+
+                NotRealizeOrRetireCommonBuildingUnit();
```
Comment (jvandaal) [inline src/BuildingRegistry/Building/Building.cs:251]: Can we move this after the loop? We know its going to be NotRealized or Retired. If we move it after the loop, we don't have to check with each iteration.

File: src/BuildingRegistry/Building/Building_State.cs [hunk 2]
```diff
@@ -78,6 +79,9 @@ private Building()
             Register<BuildingUnitWasRetiredBecauseBuildingWasDemolished>(When);
             Register<BuildingMeasurementWasChanged>(When);
 
+            Register<BuildingWasMerged>(When);
+            Register<BuildingUnitWasMoved>(When);
```
Comment (jvandaal) [inline src/BuildingRegistry/Building/Building_State.cs:83]: I'm missing the Register<BuildingUnitWasMoved> in BuildingUnit_State

File: src/BuildingRegistry/Building/Building_State.cs [hunk 3]
```diff
@@ -315,18 +328,34 @@ private void When(BuildingMeasurementWasChanged @event)
...
+            BuildingStatus = BuildingStatus.Retired;
+            _lastEvent = @event;
+        }
+
+        private void When(BuildingUnitWasMoved @event)
+        {
+            var buildingUnit =
+                BuildingUnits.Single(x => x.BuildingUnitPersistentLocalId == @event.BuildingUnitPersistentLocalId);
+            _buildingUnits.Remove(buildingUnit);
+            _lastEvent = @event;
```
Comment (jvandaal) [inline src/BuildingRegistry/Building/Building_State.cs:356]: I'm not sure if we should set the _lastEvent here? We only seem to do it for the BuildingUnitWasPlannedV2 and CommonBuildingUnitWasAddedV2 events. For the other building unit related events, we don't set the _lastEvent on Building_State.

File: src/BuildingRegistry/Building/Events/BuildingUnitWasMoved.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    using System.Linq;
+    using Be.Vlaanderen.Basisregisters.EventHandling;
+    using Be.Vlaanderen.Basisregisters.GrAr.Common;
+    using Be.Vlaanderen.Basisregisters.GrAr.Provenance;
+    using Newtonsoft.Json;
+
+    [EventTags(EventTag.For.Sync, EventTag.For.Edit, Tag.Building)]
+    [EventName(EventName)]
+    [EventDescription("De gebouweenheid werd verplaatst.")]
+    public class BuildingUnitWasMoved : IBuildingEvent
```
Comment (jvandaal) [inline src/BuildingRegistry/Building/Events/BuildingUnitWasMoved.cs:14]: We can add the interface IHasBuildingUnitPersistentLocalId

File: src/BuildingRegistry/Building/Events/BuildingWasMerged.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    using System.Linq;
+    using Be.Vlaanderen.Basisregisters.EventHandling;
+    using Be.Vlaanderen.Basisregisters.GrAr.Common;
+    using Be.Vlaanderen.Basisregisters.GrAr.Provenance;
+    using Newtonsoft.Json;
+
+    [EventTags(EventTag.For.Sync, EventTag.For.Edit, Tag.Building)]
+    [EventName(EventName)]
+    [EventDescription("Het gebouw werd samengevoegd in het doelgebouw.")]
+    public sealed class BuildingWasMerged : IBuildingEvent
```
Comment (jvandaal) [inline src/BuildingRegistry/Building/Events/BuildingWasMerged.cs:14]: This event is also defined in PR 1070 where it has different property names.

File: src/BuildingRegistry/Building/Building.cs (hunk 1 above)
Comment (ArneD) [reply src/BuildingRegistry/Building/Building.cs:251]: No, we need to respect the state of the building. From the moment one `buildingunit` remains, the common needs to disappear.

File: src/BuildingRegistry/Building/Building_State.cs (hunk 3 above)
Comment (ArneD) [reply src/BuildingRegistry/Building/Building_State.cs:356]: Exactly, when adding units the collection of the building gets altered so the `_lastEvent` is changed.
But this is also altering the collection by removing them.

But looking at the code, seems we have a bug.
`BuildingUnitWasRemovedV2`, `BuildingUnitRemovalWasCorrected` should also do this!

File: src/BuildingRegistry/Building/Events/BuildingWasMerged.cs (hunk 5 above)
Comment (ArneD) [reply src/BuildingRegistry/Building/Events/BuildingWasMerged.cs:14]: Removed from PR #1070, didn't belong there.

File: src/BuildingRegistry/Building/Building.cs (hunk 1 above)
Comment (jvandaal) [reply src/BuildingRegistry/Building/Building.cs:251]: True.

File: src/BuildingRegistry/Building/Building_State.cs (hunk 3 above)
Comment (jvandaal) [reply src/BuildingRegistry/Building/Building_State.cs:356]: @emalfroy will you add settings the _lastEvent for these events as well?

Comment (emalfroy) [reply src/BuildingRegistry/Building/Building_State.cs:379]: Fixes in another pr

**Review by pgallik — APPROVED** [review]

**Review by jvandaal — APPROVED** [review]

**Pull request merged**